frontend/catalyst/compilation_pipelines.py

```diff
@@ -795,18 +821,45 @@ def circuit(theta):
 
         from jax.core import ShapedArray
 
-            @qjit  # compilation happens at definition
-            @qml.qnode(qml.device("lightning.qubit", wires=2))
-            def circuit(x: complex, z: ShapedArray(shape=(3,), dtype=jnp.float64)):
-                theta = jnp.abs(x)
-                qml.RY(theta, wires=0)
-                qml.Rot(z[0], z[1], z[2], wires=0)
-                return qml.state()
+        @qjit  # compilation happens at definition
+        @qml.qnode(qml.device("lightning.qubit", wires=2))
+        def circuit(x: complex, z: ShapedArray(shape=(3,), dtype=jnp.float64)):
+            theta = jnp.abs(x)
+            qml.RY(theta, wires=0)
+            qml.Rot(z[0], z[1], z[2], wires=0)
+            return qml.state()
 
     >>> circuit(0.2j, jnp.array([0.3, 0.6, 0.9]))  # calls precompiled function
     array([0.75634905-0.52801002j, 0. +0.j,
            0.35962678+0.14074839j, 0. +0.j])
 
+    Catalyst also supports capturing imperative Python control flow in compiled programs. You can
+    enable this feature via the ``autograph=True`` parameter. Note that it does come with some
+    restrictions, in particular whenever global state is involved. Refer to the documentation page
+    for a complete discussion of the supported and unsupported use-cases.
+
+    .. code-block:: python
+
+        @qjit(autograph=True)
+        @qml.qnode(qml.device("lightning.qubit", wires=2))
+        def circuit(x: int):
+
+            if x < 5:
+                qml.Hadamard(wires=0)
+            else:
+                qml.T(wires=0)
+
+            return qml.expval(qml.PauliZ(0))
+
+    >>> circuit(3)
+    ...
+
+    >>> circuit(5)
+    ...
```

Comment:
I recommend including output here :)

Reply:
Oops meant to add it after I ran the example 😅

Reply:
https://github.com/PennyLaneAI/catalyst/pull/235/commits/e2334154e4720595699695d1acde4a88713a54c5